### PR TITLE
GitHub Actions: smart run strategy

### PR DESCRIPTION
**Pull request by amureki**
## Description of the Change

Previously, every pull request was running actions twice, since two signals were fired - push commit in the PR and new PR itself.

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#webhook-events

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [ ] unit-test added
- [ ] documentation updated
- [ ] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+### Fixed
```
Comment (n2ygk) [inline CHANGELOG.md:19]: Thanks for your contribution!

I don't believe this goes in the CHANGELOG though as it's not a user-visible change.

**Review by n2ygk — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (n2ygk) [reply CHANGELOG.md:19]: Oh and please add yourself to AUTHORS

**Comment by jezdez** [comment]
@amureki Please be so kind and revert this change, the use of both events is deliberate to account for all kinds of state changes a pull request can have, such as pushes happening to branches in forked repos that a pull request has been opneed against.

@n2ygk Please let me know if other pull requests come in about the GitHub Actions configuration so I can do a review. Thank you!

**Comment by amureki** [comment]
@jezdez hey Jannis! I am sorry for bringing some confusion here. ATM trying to understand which states are not covered by this configuration.
I forked the repository and created a test PR here:
https://github.com/jazzband/django-oauth-toolkit/pull/938

image


After seeing green actions build, I dropped another commit to see if it will retrigger actions event, and it did:
image

I am creating a revert PR in any case, to not possibly block anyone.

**Comment by jezdez** [comment]
> @jezdez hey Jannis! I am sorry for bringing some confusion here. ATM trying to understand which states are not covered by this configuration.
> I forked the repository and created a test PR here:
> #938
> 
> image
> 
> After seeing green actions build, I dropped another commit to see if it will retrigger actions event, and it did:
> image
> 
> I am creating a revert PR in any case, to not possibly block anyone.

Thanks for willing to dive into this, and you're right this seems to work. Sadly I don't have a working setup anymore where I was able to reproduce the edge cases with pull requests when I prepared the move from Travis to GitHub Actions. I appreciate your interest in figuring this out though!

IIRC it was a combination of pull requests being first being closed and reopened again and then not receiving push events anymore for changes in a fork branch. GitHub may have fixed it, or not, I can't really say without digging in again, their changelogs are not very detailed I'm afraid and last time the GitHub forum had a few people with similar issues. Personally I'm mostly concernd about consistency in running GitHub Actions for all Jazzband projects.

Thank you for resurfacing this since if it turns out to be not reproducable anymore I can update all the other Jazzband projects as well, which obviously would be a really nice thing to do, ecologically speaking. I'd appreciate a roll-back in the meantime though.

**Comment by amureki** [comment]
@jezdez thank you for the detailed explanation! ✨ 

Whenever you have some time and will digging into this, feel free to drop me a line here or on `amureki @ hey.com` with some links to GitHub issues or forum topics that are related to the issue.
I'd be happy then to follow up on that and try to make some test sandbox to try to reproduce different use cases, so we can move forward and possibly optimize CI runs (also do to this across all Jazzband projects).

**Pull request merged**